packages/arb-token-bridge-ui/src/components/MainContent/MainContent.tsx

```diff
@@ -27,16 +29,33 @@ export const motionDivProps = {
 }
 
 export function MainContent() {
-  const { closeTransactionHistoryPanel } = useAppContextActions()
   const { address } = useAccount()
-  const transactionHistoryProps = useTransactionHistory(address)
+  const { closeTransactionHistoryPanel } = useAppContextActions()
+  const transactionHistoryProps = useTransactionHistory(address, true)
   const {
     layout: { isTransactionHistoryPanelVisible }
   } = useAppContextState()
 
   const [isArbitrumStatsVisible] =
     useLocalStorage<boolean>(statsLocalStorageKey)
 
+  const {
+    data: { transactions },
+    updatePendingTransaction
+  } = transactionHistoryProps
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      transactions.forEach(tx => {
+        if (isTxPending(tx)) {
+          updatePendingTransaction(tx)
+        }
+      })
+    }, 10_000)
+
+    return () => clearInterval(interval)
+  }, [JSON.stringify(transactions), JSON.stringify(updatePendingTransaction)])
```

Comment:
This feels hacky 🤔 

If we had `pendingTransactions` that's a memoized version of `transactions` filtered only to use the pending ones, and then have that in the dependency array, would it solve the issue?

Reply:
updated with the suggestion